clusterloader2/pkg/measurement/common/ooms_tracker.go

```diff
@@ -163,25 +171,41 @@ func (m *clusterOOMsTrackerMeasurement) gather() ([]measurement.Summary, error)
 
 	m.stop()
 
-	content, err := util.PrettyPrintJSON(struct {
-		Ooms []oomEvent `json:"ooms"`
-	}{
-		Ooms: m.ooms,
-	})
+	oomData := make(map[string][]oomEvent)
+	oomData["failures"] = make([]oomEvent, 0)
+	oomData["past"] = make([]oomEvent, 0)
+	oomData["whitelisted"] = make([]oomEvent, 0)
+
+	for _, oom := range m.ooms {
+		if m.startTime.After(oom.Time) {
+			oomData["past"] = append(oomData["past"], oom)
+			continue
+		}
+
```

Comment:
what about using set to simplify code?
m.processWhitelist := make(map[string]bool)

if whitelisted := m.processWhitelist[oom.Process]; whitelisted {
oomData["whitelisted"] = append(oomData["whitelisted"], oom)
}

Reply:
Thanks for that remark - indeed it looks much more neat right now. Done.